sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,482 @@ def reduce_inequalities(inequalities, symbols=[]):
+    """
+    Simplex method with Bland's rule
+
+    References
+    ==========
+
+    .. [1] Thomas S. Ferguson, LINEAR PROGRAMMING: A Concise Introduction
+           web.tecnico.ulisboa.pt/mcasquilho/acad/or/ftp/FergusonUCLA_LP.pdf
+    """
+    from sympy.matrices.dense import Matrix
+
+    m, n = A.shape
+    if B.shape != (m, 1) or C.shape != (1, n):
+        raise ValueError(f"The shape of matrix B ({B.shape}) or C ({C.shape})" \
+                         f"does not match the shape of matrix A ({A.shape}).")
+
+    for val in list(A) + list(B) + list(C):
+        val = sympify(val)
+        if not (val.is_rational or isinstance(val, Float)):
+            raise TypeError(f"Only rationals and floats are allowed in the Simplex method." \
+                             f" ({val}) is neither.")
+
+    D = ImmutableMatrix([0])
+    M = Matrix([[A, B], [-C, D]])
+
+    r_orig = ['x_{}'.format(j) for j in range(M.cols - 1)]
+    s_orig = ['y_{}'.format(i) for i in range(M.rows - 1)]
+
+    R = r_orig.copy()
+    S = s_orig.copy()
+
+    # Phase 1: find a feasible solution or determine none exist
+    while True:
+        B = M[:-1, -1]
+        A = M[:-1, :-1]
+        if all(B[i] >= 0 for i in range(B.rows)):
+            # We have found a feasible solution
+            break
+
+        # Find k, first row with a negative rightmost entry
+        for k in range(B.rows):
+            if B[k] < 0:
+                break
+
+        # Choose pivot column, j0
+        piv_cols = [j for j in range(A.cols) if A[k, j] < 0]
+        if not piv_cols:
+            raise InfeasibleLinearProgrammingError('The constraint set is empty!')
+        j0 = sorted(piv_cols, key=lambda c: R[c])[0] # Bland's rule
+
+        # Choose pivot row, i0
+        piv_rows = [i for i in range(A.rows) if A[i, j0] > 0 and B[i] > 0]
+        piv_rows.append(k)
+        i0 = _choose_pivot_row(A, B, piv_rows, j0, S)
+
+        M = _pivot(M, i0, j0)
+        R[j0], S[i0] = S[i0], R[j0]
+
+    # Phase 2: starting at a feasible solution, pivot until we reach optimal solution
+    if not skip_phase_2:
+        while True:
+            B = M[:-1, -1]
+            A = M[:-1, :-1]
+            C = M[-1, :-1]
+
+            # Choose a pivot column, j0
+            piv_cols = []
+            piv_cols = [j for j in range(C.cols) if C[j] < 0]
+            if not piv_cols:
+                break
+            j0 = sorted(piv_cols, key=lambda c: R[c])[0] # Bland's rule
+
+            # Choose a pivot row, i0
+            piv_rows = [i for i in range(A.rows) if A[i, j0] > 0]
+            if not piv_rows:
+                raise UnboundedLinearProgrammingError('Objective function can assume arbitrarily large values!')
+            i0 = _choose_pivot_row(A, B, piv_rows, j0, S)
+
+            M = _pivot(M, i0, j0)
+            R[j0], S[i0] = S[i0], R[j0]
+
+    argmax = [None]*(M.cols-1)
+    argmin_dual = [None]*(M.rows-1)
+
+    for i, var in enumerate(R):
+        v = var[0]
```

Comment:
I am not seeing where an element or R (or S) becomes an indexed object after being created as a string at 1106-1110.

Reply:
I'm using var[0] to get the first character of the string which will be either 'x' or 'y'. Is this bad?

Reply:
Previous versions of the code used a double nested for loops which I thought was unintuitive and slower.

Reply:
It would be better not to use strings. The lists R and S can just store something like tuples of bools and integers.

Reply:
I'm using tuples of a string and integer now. Like ('x', 1) or ('y', 1). Would it be better to use a bool instead of 'x' and 'y'?

Reply:
Yeah I guess that's probably better. I'll change it.